src/server/routes/entity/entity.tsx

```diff
@@ -481,7 +483,15 @@ export function handleDelete(
 		return savedMainEntity.toJSON({omitPivot: true});
 	});
 
-	return handler.sendPromiseResult(res, entityDeletePromise, search.deleteEntity);
+	try {
+		res.send(entityDelete);
+		search.deleteEntity(entityDelete);
+		return entityDelete;
+	}
+	catch (err) {
+		log.error(err);
+		return error.sendErrorAsJSON(res, err);
+	}
```

Comment:
Here there is a small subtlety which is that previously search.deleteEntity was called after the main entityDeletePromise succeeded, in such a way that an error thrown in search.deleteEntity was not sent back to the user.
Since we are now awaiting the promise further up, we don't need this try-catch block for returning the response (res.send) anymore, but we do need one for separately catching search indexing errors like so:

 ```suggestion
	res.send(entityDelete);
	try {
		search.deleteEntity(entityDelete);
	}
	catch (err) {
		log.error(err);
	}
	return entityDelete;
```


However, we might need to wrap all the code from `const entityDelete = await ...` onwards into another top-level try-catch block, and in the catch block use that `return error.sendErrorAsJSON(res, err);` .

Reply:
done 👍

 (〃￣︶￣)人(￣︶￣〃)